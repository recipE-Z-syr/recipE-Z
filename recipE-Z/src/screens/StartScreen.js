import React, { Component } from 'react';
import { Button, StyleSheet, Text, TextInput, View, SafeAreaView, ScrollView, Keyboard, TouchableOpacity, Image } from 'react-native';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

class StartScreen extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {navigation} = this.props;
    return (
        <SafeAreaView style={styles.container}>
          <View style={{alignItems: 'center', justifyContent: 'center', backgroundColor:'#ffcece' }}>
          <View style={{width: 334, height: 561, backgroundColor: 'white', position: 'absolute'}}/>
              <Image
                style={{width: 155, height: 150, top:-25}}
                source={require('../img/logo1.png')}
              />
              <Text style={{color: '#ed4848', fontSize: 24, fontWeight: 'bold'}}>
              Welcome to recipE-Z
              </Text>
              <Text>{"     "}</Text>
              <Text>{"     "}</Text>
              <Text>
              Don’t know what to cook?
              </Text>
              <Text>
              Have stuff you don’t want to go to waste?
              </Text>
              <Text>
              Just enter the ingredients you have
              </Text>
              <Text>
               and we’ll help you find a recipe.
              </Text>
              <Text>{"     "}</Text>
              <Text>{"     "}</Text>
              <Text style={{color: '#ed4848', fontSize: 16, fontWeight: 'bold'}}>
              We put the E-Z in recipE-Z
              </Text>
              <Text>{"     "}</Text>
              <Text>{"     "}</Text>
              <TouchableOpacity
                style={styles.startButton}
                onPress={() => navigation.navigate('Log-In')}
                underlayColor='#ed4848'>
                <Text style={styles.startText}>LET'S GET STARTED</Text>
              </TouchableOpacity>
        </View>
        </SafeAreaView>
   ); //end return
 }//end render
}//end class

export default StartScreen;

const styles = StyleSheet.create({
});
