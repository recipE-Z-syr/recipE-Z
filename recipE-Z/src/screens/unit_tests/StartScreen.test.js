//StartScreen.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, ShallowWrapper } from "enzyme";

import StartScreen from '../StartScreen';

describe('<StartScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<StartScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
});

test('Navigates to Sign-Up', () => {
  // const spy = jest.spyOn(navigation, 'navigate')
  let wrapper: ShallowWrapper;
  let props: any
  props = createTestProps();
  wrapper = shallow(<StartScreen {...props} />);  

  const startButton = wrapper.find('TouchableOpacity').at(0)
  startButton.props().onPress()

  expect(props.navigation.navigate).toBeCalledWith('Sign-Up')
})