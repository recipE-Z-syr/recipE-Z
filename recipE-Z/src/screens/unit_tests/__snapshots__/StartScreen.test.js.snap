// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StartScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "fontFamily": "nunito-regular",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#ffcece",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "height": 561,
          "justifyContent": "space-around",
          "paddingBottom": 70,
          "paddingTop": 70,
          "position": "absolute",
          "width": 334,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 150,
            "marginBottom": 10,
            "width": 155,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#ed4848",
            "fontFamily": "nunito-extrabold",
            "fontSize": 20,
          }
        }
      >
        Welcome to recipE-Z
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "nunito-regular",
            }
          }
        >
          Don’t know what to cook?
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "nunito-regular",
            }
          }
        >
          Have stuff you don’t want to go to waste?
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "nunito-regular",
            }
          }
        >
          Just enter the ingredients you have
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "nunito-regular",
            }
          }
        >
          and we’ll help you find a recipe.
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "#ed4848",
            "fontFamily": "nunito-bold",
            "fontSize": 16,
          }
        }
      >
        We put the E-Z in recipE-Z
      </Text>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ed4848",
            "borderColor": "#ed4848",
            "borderRadius": 5,
            "borderWidth": 10,
            "justifyContent": "center",
            "marginTop": 15,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "nunito-bold",
              "paddingLeft": 10,
              "paddingRight": 10,
              "textAlign": "center",
            }
          }
        >
          LET'S GET STARTED
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
